### ISSUE TITLE
Move cookiepassword to a session for security

### ISSUE THREAD

**Issue by RlndVt** [causefx/Organizr#116]
#  Background:

As a workaround for HTTP Proxy #20 @ImperialXT suggested adding the bold line (or something similar) to your proxy configuration in nginx:

<pre>
 location /service {
    <b>if ($cookie_cookiePassword != "PASSWORDHERE") { return 401; }</b>
    add_header X-Frame-Options "SAMEORIGIN";
    proxy_pass http://IP:PORT;
 }
</pre>

This only allows access to the reverse proxies if a cookie, created by logging in to organizr, is set that matches. 

# The Problem:

In the current state if someone gains possession of an active cookie (forgetting to logout or some malicious program) they can access the reverse proxies. The cookie is only destroyed when you logout, when it has timed out (possibly, untested), or miss matches when accessing your organizr site. If it doesn't get destroyed there is unfettered access to the proxies.

# (Possible) Solution:

Move the cookiePassword to a session. That way it is destroyed when the windows is closed.

**Comment by elRadix** [comment]
i'm using a sub domain on a nginx "webserver tabs.example.com", how can I use the cookie password without a proxy?

**Comment by ImperialXT** [comment]
@RlndVt as far as I'm aware there's no easy way to access PHP session data from nginx. There may be extensions that do it, but the default install of nginx does not support this nor does nginx.

I did consider this when I first came up with the idea.

**Comment by ImperialXT** [comment]
@elRadix it should be easy enough, but I'd need to know what exactly your setup is like

**Comment by elRadix** [comment]
This is my config, maybe not the best one. Happy if some can improve it :)

> server {
>  listen 443 ssl http2 deferred;
>  server_name tabs.example.com;
> 
>  root /var/www/organizr;
> 
>
>  index index.php index.html index.htm index.nginx-debian.html;
> 
>  
>  auth_basic "This site is Restricted";
>  auth_basic_user_file /etc/nginx/.htpasswd;
> 
>  access_log /var/log/nginx/access.log;
>  error_log    /var/log/nginx/organizr.error.log;
> 
> 
>  >  error_page 401 403 404 /404.html;
> 
>  
>  location / {
>           try_files $uri $uri/ =404;
>  }
>  
>  location ~ /\.ht {
>           deny all;
> 
>   ssl on;
>   include /etc/nginx/snippets/strong-ssl.conf;
>   ssl_certificate /etc/letsencrypt/live/www.example.com/fullchain.pem;
>   ssl_certificate_key /etc/letsencrypt/live/www.example.com/privkey.pem;
> 
>     index index.php index.html index.htm;
>     autoindex on;
>     try_files $uri $uri/ /index.php;
>     client_max_body_size 10m;
>     client_body_buffer_size 8K;
>     client_header_buffer_size 1k;
>     large_client_header_buffers 4 8k;
> 
>         location ~ \.php$ {
>                 fastcgi_split_path_info ^(.+\.php)(/.+)$;
>                 fastcgi_pass unix:/var/run/php5-fpm.sock;
>                 fastcgi_index index.php;
>                 include fastcgi.conf;
>         }
> 
> }
>

**Comment by ImperialXT** [comment]
@elRadix unfortunately the cookie thing wont work for you. The cookie thing is being used to prevent access to things such as Sonar or Sab etc without someone going through Organizr itself.

Putting the cookie check on your oragnizr vhost would break things as you wouldn't be able to access organizr itself.

**Comment by causefx** [comment]
@elRadix is that your whole config?  you arent reverse proxying other apps?  you wouldn't do it for orgaznir, only the other services.

**Comment by rix1337** [comment]
Can you eli5 the cookie password thing?

Do I just set that password in every reverse proxy block of my nginx config?

**Comment by causefx** [comment]
@rix1337 exactly, you put that in the reverse proxy block and it will check with Organizr if it is supposed to have access.

**Comment by causefx** [comment]
looks like this will not be feasible.  I will look for alternatives.
